### ISSUE TITLE
How to handle boolean fields?

### ISSUE THREAD

**Issue by Siceberg** [t-tiger/gorm-bulk-insert#43]
This is my test example:

```
type Test struct {
	Name     string `gorm:"Column:name;Size:100;DEFAULT:"`
	IsAdmin  bool   `gorm:"Column:is_admin;Size:1;DEFAULT:false"`
}
```

When I use gorm Create() function

```
db.Create(&Test{
	Name: "n1",
	IsAdmin: false,
})
```

I will get sql like:

`INSERT INTO `test` (`name`,`is_admin`) VALUES ('n1','0')`

When I use gormbulk.BulkInsert()

```
gormbulk.BulkInsert(
	db, []interface{}{
		Test{
			Name: "n1",
			IsAdmin: false,
		},
	}, 3000,
)
```

I will get sql like:

`INSERT INTO `test` (`name`, `is_admin`) VALUES ('n1','false')`

How should I deal with boolean fields to get results similar to gorm Create()?

**Comment by bombsimon** [comment]
TL;DR: Don't use `false` as default value in your struct tag if using MySQL. Remove the default value or set it to `0`.

Everything except size is the default values so you could define your type like this:

```go
type Test struct {
	Name    string `gorm:"size:100"` // Or `gorm:"type:varchar(100)"`
	IsAdmin bool
}
```
---

Postgres has a boolean type which can be set to real true or false but MySQL uses `TINYINT(1)` as boolean storage. Newer versions of MySQL has the keyword boolean to define tables but they still store true as `1` and false as `0`.

As for your example, I cannot get your definition to auto migrate with the empty default value. If I add an empty string and run `gorm.AutoMigrate(&Test{})` I get the following schema:

```sql
+----------+--------------+------+-----+---------+-------+
| Field    | Type         | Null | Key | Default | Extra |
+----------+--------------+------+-----+---------+-------+
| name     | varchar(100) | YES  |     |         |       |
| is_admin | tinyint(1)   | YES  |     | 0       |       |
+----------+--------------+------+-----+---------+-------+
```

The actual SQL created by this package will result in a prepared statement, something like this:

```sql
INSERT INTO `tests` (`is_admin`, `name`) VALUES (?, ?), (?, ?)
```

The bound values will be whatever you have in your slice. These values are sent to `gorm.Exec()` and that's where the handling of the values are being made.

If I run the code with your tags, I get a similar error like you, `Error 1366: Incorrect integer value: 'false' for column 'is_admin' at row 1`.  It's related to your default value `false` which isn't compatible with how `gorm` handles your query. Since `false` is the default value for a boolean column you can just remove it. If you for some reason would like to keep it explicitly, change it to `0` instead of `false` (for MySQL).

<details><summary>Fully working example creating table and inserting booleans</summary>

```go
package main

import (
	"database/sql"
	"fmt"
	"log"
	"os"

	"github.com/jinzhu/gorm"
	gormbulk "github.com/t-tiger/gorm-bulk-insert"

	_ "github.com/jinzhu/gorm/dialects/mysql"
)

const (
	// docker run --rm -it -e MYSQL_ALLOW_EMPTY_PASSWORD=1 -p 3306:3306  mysql:latest
	testDB  = "test_db"
	testDSN = "root:@/" + testDB + "?parseTime=1"
)

// Test is for testing.
type Test struct {
	Name    string `gorm:"Column:name;Size:100;DEFAULT:''"`
	IsAdmin bool   `gorm:"Column:is_admin;Size:1;DEFAULT:0"`
}

func main() {
	var (
		sqlDB, _ = sql.Open("mysql", testDSN)
		db, _    = gorm.Open("mysql", testDSN)
		logger   = log.New(os.Stdout, "[DB DEBUG] ", log.LstdFlags)
	)

	db.LogMode(true)
	db.SetLogger(logger)

	defer db.Close()

	initCommands := []string{
		fmt.Sprintf("DROP DATABASE IF EXISTS `%s`", testDB),
		fmt.Sprintf("CREATE DATABASE `%s`", testDB),
	}

	for _, q := range initCommands {
		if _, err := sqlDB.Exec(q); err != nil {
			panic(err)
		}
	}

	// Create base table
	db.AutoMigrate(&Test{})

	toInsert := []interface{}{
		Test{Name: "first", IsAdmin: false},
		Test{Name: "second", IsAdmin: true},
	}

	if err := gormbulk.BulkInsert(db, toInsert, 3000); err != nil {
		panic(err)
	}
}
```
</details>

**Comment by Siceberg** [comment]
@bombsimon 
Thank you very much for your reply, it is very helpful to me
